File: Backtrack.Core/Backtrack.Core.WebApi/Contracts/Common/ApiResponse.cs
namespace Backtrack.Core.WebApi.Contracts.Common;

public sealed record ApiResponse<T>
{
    public required bool Success { get; init; }
    public T? Data { get; init; }
    public ApiError? Error { get; init; }
    public required string CorrelationId { get; init; }

    public static ApiResponse<T> SuccessResponse(T data, string correlationId)
    {
        return new ApiResponse<T>
        {
            Success = true,
            Data = data,
            Error = null,
            CorrelationId = correlationId
        };
    }

    public static ApiResponse<T> ErrorResponse(ApiError error, string correlationId)
    {
        return new ApiResponse<T>
        {
            Success = false,
            Data = default,
            Error = error,
            CorrelationId = correlationId
        };
    }
}

public sealed record ApiError
{
    public required string Code { get; init; }
    public required string Message { get; init; }
    public object? Details { get; init; }
}

File: Backtrack.Chat/src/utils/api-response.ts
export interface ApiResponse<T> {
  success: boolean;
  data?: T;
  error?: ApiError;
  correlationId?: string;
}

/**
 * API Error structure
 */
export interface ApiError {
  code: string;
  message: string;
  details?: unknown;
}

/**
 * Helper class to build API responses
 */
export const ApiResponseBuilder = {
  success<T>(data: T, correlationId?: string): ApiResponse<T> {
    return {
      success: true,
      data,
      ...(correlationId && { correlationId }),
    };
  },

  error<T>(error: ApiError, correlationId?: string): ApiResponse<T> {
    return {
      success: false,
      error,
      ...(correlationId && { correlationId }),
    };
  },
} as const;
